samples/dotnet/openapi-skills/Program.cs

```diff
@@ -0,0 +1,229 @@
+            .WithLogger(loggerFactory.CreateLogger<IKernel>())
+            .Configure(c =>
+            {
+                switch (aiOptions.Type)
+                {
+                    case AIServiceOptions.AIServiceType.AzureOpenAI:
+                        c.AddAzureChatCompletionService(aiOptions.Models.Completion, aiOptions.Endpoint, aiOptions.Key);
+                        break;
+                    case AIServiceOptions.AIServiceType.OpenAI:
+                        c.AddOpenAIChatCompletionService(aiOptions.Models.Completion, aiOptions.Key);
+                        break;
+                    default:
+                        throw new InvalidOperationException($"Unhandled AI service type {aiOptions.Type}");
+                }
+            })
+            .Build();
+
+
+        // Register the GitHub skill using an OpenAPI definition containing only pull request GET operations.
+        GitHubSkillOptions gitHubOptions = configuration.GetRequiredSection(GitHubSkillOptions.PropertyName).Get<GitHubSkillOptions>()
+            ?? throw new InvalidOperationException($"Missing configuration for {GitHubSkillOptions.PropertyName}.");
+
+        BearerAuthenticationProvider authenticationProvider = new(() => Task.FromResult(gitHubOptions.Key));
+
+        await kernel.ImportOpenApiSkillFromFileAsync(
+            skillName: "GitHubSkill",
+            filePath: Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)!, "GitHubSkill/openapi.json"),
+            authCallback: authenticationProvider.AuthenticateRequestAsync);
+
+
+        // Use a planner to decide when to call the GitHub skill. Since we are not chaining operations, use
+        // the ActionPlanner which is a simplified planner that will always return a 0 or 1 step plan.
+        ActionPlanner planner = new(kernel, logger: logger);
+
+        // Chat loop
+        IChatCompletion chatGPT = kernel.GetService<IChatCompletion>();
+        OpenAIChatHistory chatHistory = (OpenAIChatHistory)chatGPT.CreateNewChat("You are a helpful, friendly, intelligent assistant that is good at conversation.");
+        while (true)
+        {
+            Console.WriteLine("----------------");
+            Console.Write("Input: ");
+            string? input = Console.ReadLine();
+
+            if (string.IsNullOrWhiteSpace(input))
+            {
+                continue;
+            }
+
+            // Add GitHub's response, if any, to the chat history.
+            int planResultTokenAllowance = (int)(aiOptions.TokenLimit * 0.25); // Allow up to 25% of our token limit to be from GitHub.
+            string planResult = await PlanGitHubSkill(gitHubOptions, planner, chatHistory, input, planResultTokenAllowance);
+            if (!string.IsNullOrWhiteSpace(planResult))
+            {
+                chatHistory.AddUserMessage(planResult);
+            }
+
+            // Add the user's input to the chat history.
+            chatHistory.AddUserMessage(input);
```

Comment:
Shouldn't this be added before the plan??

Reply:
The plan is adding context to user's inquiry, so we want the plan result to come before the ask.